modules/tinymce/src/core/main/ts/content/ContentTypes.ts

```diff
@@ -10,23 +10,42 @@ import AstNode from '../api/html/Node';
 export type Content = string | AstNode;
 export type ContentFormat = 'raw' | 'text' | 'html' | 'tree';
 
-export interface GetContentArgs {
-  format?: ContentFormat;
-  get?: boolean;
-  content?: string;
-  getInner?: boolean;
+interface GetContentArgsBase {
+  format: ContentFormat;
+  get: boolean;
+  getInner: boolean;
   no_events?: boolean;
   [key: string]: any;
 }
 
-export interface SetContentArgs {
-  format?: string;
-  set?: boolean;
-  content?: string;
+interface SetContentArgsBase {
+  format: string;
+  set: boolean;
+  content: Content;
   no_events?: boolean;
   no_selection?: boolean;
 }
 
+export interface GetContentArgs extends GetContentArgsBase {
+  content: Content;
+}
+
+export interface SetContentArgs extends SetContentArgsBase {
+  content: Content;
+}
+
+export interface SelectionSetContentArgs extends SetContentArgsBase {
+  content: string;
+  selection?: boolean;
+  contextual?: boolean;
+}
+
+export interface GetSelectionContentArgs extends GetContentArgsBase {
+  selection?: boolean;
```

Comment:
```suggestion
  selection: boolean;
```
This one is always defined as per `GetSelectionContentImpl`.

Reply:
Woups, good catch.